openquake/hazardlib/gsim/kanno_2006.py

```diff
@@ -0,0 +1,517 @@
+
+from __future__ import division
+import warnings
+import numpy as np
+from scipy.constants import g
+from openquake.hazardlib import const, imt
+from openquake.hazardlib.gsim.base import GMPE, CoeffsTable
+# from openquake.hazardlib.gsim.si_midorikawa_1999 \
+#    import _get_min_distance_to_sub_trench
+
+
+class Kanno2006Shallow(GMPE):
+    # pylint: disable=too-few-public-methods
+    """
+    Implements GMPE of Kanno et al. (2006) for shallow events based on data
+    predominantly from Japan.
+
+    Note that "both crustal and subduction interface events fall into the
+    category of shallow events" (p. 883) where "shallow" is defined as "focal
+    depth of 30 km or less" (p. 895).
+
+    Verification of mean value data was done by digitizing Figures 4 and 5
+    using http://arohatgi.info/WebPlotDigitizer/ app/. The maximum error was
+    15% while the average error was 3-4%.
+
+    Page number citations in this documentation refer to:
+
+    Kanno, T., Narita, A., Morikawa, N., Fujiwara, H., and Fukushima, Y.
+    (2006). A new attenuation relation for strong ground motion in Japan based
+    on recorded data. Bull. Seism. Soc. Am. 96(3):879–897.
+    """
+
+    #: This model is generally considered to be intended for subduction
+    #: regions, but the authors do not constrain the type of event, only the
+    #: depth: "both crustal and subduction interface events fall into the
+    #: category of shallow events." (p. 883)
+    DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.SUBDUCTION_INTERFACE
+
+    #: "regression coefficients for the base model in equations (5) and (6)
+    #: for PGA , PGV , and 5% damped response spectral acceleration are given"
+    #: (p. 883)
+    DEFINED_FOR_INTENSITY_MEASURE_TYPES = set([imt.PGA, imt.PGV, imt.SA])
+
+    #: "The peak value is the peak square root of the sum of squares
+    #: of two orthogonal horizontal components in the time domain" (p. 880)
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.PEAK_SRSS_HORIZONTAL
+
+    #: Although interevent and intraevent residuals are separately discussed
+    #: in the context of focal depth and site conditions, only the total
+    #: standard deviation is tabulated.
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = set([const.StdDev.TOTAL])
+
+    #: "Coefficients p and q were derived by regression analysis on the
+    #: residuals averaged at intervals of every 100 m/sec in AVS30." (p. 884)
+    REQUIRES_SITES_PARAMETERS = set(('vs30',))
+
+    #: Sole required rupture parameter is magnitude; faulting style is not
+    #: addressed.
+    REQUIRES_RUPTURE_PARAMETERS = set(('mag',))
+
+    #: "The source distance is the closest distance from a fault plane to the
+    #: observation site and is the hypocentral distance in the case of
+    #: earthquakes for which the fault model is not available." (p. 880)
+    REQUIRES_DISTANCES = set(('rrup',))
+
+    def get_mean_and_stddevs(self, sites, rup, dists, im_type, stddev_types):
```

Comment:
Please keep the naming convention the same as for other GMPEs. `im_type` should be `imt`